File: src/cpp/rick.cpp
#include <iostream>

class Rick
{
    public:
        Rick never()
        {
            std::cout << "Never ";
            return *this;
        }

        Rick gonna()
        {
            std::cout << "gonna ";
            return *this;
        }

        Rick give()
        {
            std::cout << "let ";
            return *this;
        }

        Rick you()
        {
            std::cout << "you ";
            return *this;
        }

        void up()
        {
            std::cout << "down!" << std::endl;
        }
};

int main() {
    Rick rick;
    rick.never().gonna().give().you().up();
    return 0;
}

File: src/kt/rick.kt
class Rick {
    fun never(): Rick {
        print("Never ")
        return this
    }

    fun gonna(): Rick {
        print("gonna ")
        return this
     }

    fun give(): Rick {
        print("let ")
        return this
    }

    fun you(): Rick {
        print("you ")
        return this
    }

    fun up() = println("down!")
}

fun main() {
    val rick = Rick()
    rick.never().gonna().give().you().up()
}
